pkg/controller/bookkeepercluster/bookkeepercluster_controller.go

```diff
@@ -265,7 +265,14 @@ func (r *ReconcileBookkeeperCluster) reconcileFinalizers(bk *bookkeeperv1alpha1.
 				return fmt.Errorf("failed to update Bookkeeper object (%s): %v", bk.Name, err)
 			}
 			if err = r.cleanUpZookeeperMeta(bk, pravegaClusterName); err != nil {
-				return fmt.Errorf("failed to clean up metadata (%s): %v", bk.Name, err)
+				// emit an event for zk metadata cleanup failure
+				message := fmt.Sprintf("failed to clean up metadata (%s): %v", bk.Name, err)
+				event := util.NewEvent("CLEANUP_ERROR", bk, "ZK Metadata Cleanup Failed", message, "Error")
```

Comment:
Could we call this ZKMETA_CLEANUP_ERROR... or something similar and also publish the path???